### ISSUE TITLE
Appsync with NodejsFunction lambda resolver doesn't work with ESM format

### ISSUE THREAD

**Issue by csbszb** [aws/aws-cdk#19210]
### What is the problem?

I have an Appsync API with a lambda resolver in nodejs, created with the NodejsFunction function. When changing the bundling option to `format: OutputFormat.ESM` I see the following error in the Appsync logs:
`"error": {
            "message": "index.main is undefined or not exported",
            "type": "Runtime.HandlerNotFound"
},`

When I remove the format option, everything is working fine again.

### Reproduction Steps

1. Create a GraphqlApi
2. Create a lambda function with NodejsFunction and set the bundling option `format: OutputFormat.ESM`
3. Add the lambda function as a resolver.

### What did you expect to happen?

Appsync executes the queries without any errors just like when the format option wasn't set

### What actually happened?

Got the following error:

`"error": {
            "message": "index.main is undefined or not exported",
            "type": "Runtime.HandlerNotFound"
},`

### CDK CLI Version

2.15.0 (build 151055e)

### Node.js Version

14

### OS

macOS 12.1

### Language

Typescript

**Comment by otaviomacedo** [comment]
@jogold do you have any idea what's going on here?

**Comment by jogold** [comment]
@csbszb so you're exporting a `main` function in your code?

What happens when you test your function in the Lambda console?

**Comment by csbszb** [comment]
@jogold yes, I'm exporting the `main` function as the handler. When the ESM output format is enabled in CDK and I invoke the lambda directly from the console, I receive the result without any errors.

With ESM the generated `index.mjs` file looks like this: 
```javascript
var main = async (event) => {
  ...
};
export {
  main
};
```

**Comment by jogold** [comment]
Since the Lambda function can be successfully tested in "isolation" could this be a problem between AppSync and Lambda services? Not CDK related then?

**Comment by csbszb** [comment]
I would say it's more about how CDK generates/bundles the lambda code whether the `format: OutputFormat.ESM` option is set or not, given the same exact lambda function:
- when `format: OutputFormat.ESM` is set -> AppSync calls run with error though direct lambda invocation works without any error
- when format is not set, or `format: OutputFormat.CJS` option is used -> AppSync calls runs without any error

**Comment by csbszb** [comment]
@otaviomacedo, @jogold : is there anything I can help you with to sort this out?

@jogold : based on your comment am I right to assume that Appsync currently cannot call Lambda resolvers that are bundled with ESM?

**Comment by jogold** [comment]
> @jogold : based on your comment am I right to assume that Appsync currently cannot call Lambda resolvers that are bundled with ESM?

Yes, this is my understanding. Something in the AppSync Service. I would open a support case for this.
